### PR TITLE
link to archived pages is not a broken link

### PR DESCRIPTION
**Pull request by peterbe**
Part of #2675

I went down a rabbit hole. Aka. I fell down a trap rabbit hole. 
I did some poking around in what it would take to fix these archived broken links. I wrote some code to convert the `archived.txt` into URLs and stuff. Now, I had gone all this way to figure out what's going on and I have all this code. So I decided to keep it. At least it solves the problem that some broken links no longer appear as broken if they are just links to the archived content. It felt cheap and safe to do. 

Before:
Screen Shot 2021-01-25 at 4 24 17 PM

After:
Screen Shot 2021-01-25 at 4 25 43 PM
*Gone!*

So what's [left is all the busted old archived redirect URLs](https://github.com/mdn/yari/issues/2675#issuecomment-767124481). We *could* recover these and create a file like `archived-redirects.txt`. But I'm not sure it's super worth it. I've already spent too much time on this PR :)

### CODE REVIEW

**Comment by peterbe** [comment]
By the way http://localhost:3000/en-US/docs/Mozilla/Firefox#_flaws is the page I used for my screenshots. 
But look at pages like http://localhost:3000/en-US/docs/Mozilla/Developer_guide#_flaws which is full of broken link flaws now because we never migrated the redirects.

File: content/redirect.js [hunk 1]
```diff
@@ -11,7 +11,37 @@ const {
...
+
+function getArchivedURLs() {
+  if (!ARCHIVED_URLS) {
+    ARCHIVED_URLS = new Set();
+    for (const folder of getArchivedPaths()) {
+      // We don't need to use `path.sep` here because all the files in the
+      // archived.txt file are written explicitly with `/`.
+      const split = folder.split("/");
+      split.unshift("");
+      split.splice(2, 0, "docs");
```
Comment (fiji-flo) [inline content/redirect.js:26]: paths with a `_colon_` cannot be safely reverted to an URL. Either we hack this in here an are well aware of it not being 100% accurate or we stick with converting urls to paths and compare those.

**Review by fiji-flo — CHANGES_REQUESTED** [review]

File: content/redirect.js (hunk 1 above)
Comment (peterbe) [reply content/redirect.js:26]: Good catch! I can't believe I missed that. Yeah, we need the reverse of slugToFoldername (or whatever it's called).

Comment (peterbe) [reply content/redirect.js:26]: Ugh. That reminds me, we don't have a reverse of `slugToFolder` :(
I could write a function that does:
```
   .replace(/_doublecolon_/g, '::')
```
etc.
But that doesn't take care of whatever `sanitizeFilename` might do. That one is not mapping 1-to-1. 
Also, we'd lose the casing. But that might not be a great solution. 
What we should perhaps do is to rewrite `archived.txt` to be a list of URLs instead. 
What do you think about that?
Last time I wrote a python script that traversed all the `index.html` files in mdn/archived-content and extracted the slug.

Comment (peterbe) [reply content/redirect.js:26]: We chatted about it on Zoom and concluded that the simplest solution is to do this:
```diff
function isArchivedURL(url) {
- return getArchivedURLs().has(url)
+ return getArchivedFilepaths().has(slugToFoldername(url))  
}
```

**Review by fiji-flo — APPROVED** [review]
👍

**Pull request merged**